Polyhedron/demo/Polyhedron/Scene_c3t3_item.cpp

```diff
@@ -1483,11 +1484,23 @@ Scene_c3t3_item::setColor(QColor c)
 void Scene_c3t3_item::show_grid(bool b)
 {
   d->is_grid_shown = b;
+  Q_FOREACH(QAction* action, contextMenu()->actions())
+    if(action->objectName() == "actionShowGrid")
+    {
+      action->setChecked(b);
+      break;
+    }
   itemChanged();
```

Comment:
Use `contextMenu.findChild<QAction>("actionShowGrid")` to retrieve the action, instead of searching with a loop.